### PR TITLE
Add instanced drawing support for AmberScript

### PR DESCRIPTION
**Pull request by asuonpaa**
Fixes https://github.com/google/amber/issues/565

### CODE REVIEW

File: docs/amber_script.md [hunk 1]
```diff
@@ -608,10 +609,14 @@ RUN {pipeline_name} \
 # data must be attached to the pipeline.
 
 # A start index of |value| will be used and the count of |count_value| items
-# will be processed.
+# will be processed. The draw is instanced if |inst_count_value| is greater
+# than one. In case of instanced draw |inst_value| controls the starting
+# instance ID.
 RUN {pipeline_name} DRAW_ARRAY AS {topology} \
     [ START_IDX _value_ (default 0) ] \
     [ COUNT _count_value_ (default vertex_buffer size - start_idx) ]
```
Comment (dj2) [inline docs/amber_script.md:617]: I think this line needs a `\` on the end.

File: docs/amber_script.md [hunk 2]
```diff
@@ -620,10 +625,14 @@ RUN {pipeline_name} DRAW_ARRAY AS {topology} \
 # drawn using the given |topology|.
 #
 # A start index of |value| will be used and the count of |count_value| items
-# will be processed.
+# will be processed. The draw is instanced if |inst_count_value| is greater
+# than one. In case of instanced draw |inst_value| controls the starting
+# instance ID.
 RUN {pipeline_name} DRAW_ARRAY AS {topology} INDEXED \
     [ START_IDX _value_ (default 0) ] \
     [ COUNT _count_value_ (default index_buffer size - start_idx) ]
```
Comment (dj2) [inline docs/amber_script.md:633]: ditto

File: src/amberscript/parser.cc [hunk 3]
```diff
@@ -2356,37 +2356,36 @@ Result Parser::ParseRun() {
...
       token = tokenizer_->NextToken();
 
-      if (!token->IsEOS() && !token->IsEOL()) {
-        if (!token->IsIdentifier() || token->AsString() != "COUNT")
-          return Result("missing COUNT for RUN command");
+      if (!token->IsIdentifier())
+        return Result("expecting identifier for RUN command");
+
+      if (token->AsString() == "INDEXED") {
+        if (!pipeline->GetIndexBuffer())
```
Comment (dj2) [inline src/amberscript/parser.cc:2374]: nit: add `{}`'s since the body is multiple lines.

File: docs/amber_script.md (hunk 1 above)
Comment (asuonpaa) [reply docs/amber_script.md:617]: done

File: src/amberscript/parser.cc (hunk 3 above)
Comment (asuonpaa) [reply src/amberscript/parser.cc:2374]: done

**Pull request merged**